packages/pluggableWidgets/line-chart-native/src/components/LineChart.tsx

```diff
@@ -44,15 +44,39 @@ export function LineChart(props: LineChartProps): ReactElement | null {
 
     const dataTypesResult = useMemo(() => getDataTypes(series), [series]);
 
+    // Line Chart user-styling may be missing for certain series. A palette is passed, any missing line colours
+    // fallback to a colour from the palette.
+    const normalisedLineColors: string[] = useMemo(() => {
+        const result: string[] = [];
+        let index = 0;
+
+        for (const _series of series) {
+            const configuredStyle = !_series.customLineStyle
+                ? null
+                : style.lineStyles?.[_series.customLineStyle]?.line?.data?.stroke;
+
+            if (typeof configuredStyle !== "string") {
+                result.push(style.lineColorPalette?.[index] || "black");
+                if (style.lineColorPalette) {
+                    index = index + 1 === style.lineColorPalette.length ? 0 : index + 1;
+                }
+            } else {
+                result.push(configuredStyle);
+            }
+        }
+
+        return result;
```

Comment:
Then just return the map or not even define the block `{}`

Reply:
We still need the block for the lineColorPalette index tracking